src/Elements/Floor.cs

```diff
@@ -64,55 +65,55 @@ public List<Opening> Openings
         /// <summary>
         /// Create a floor.
         /// </summary>
-        /// <param name="profile">The profile of the floor.</param>
+        /// <param name="perimeter">The floor perimeter.</param>
         /// <param name="elementType">The floor type of the floor.</param>
         /// <param name="elevation">The elevation of the top of the floor.</param>
         /// <param name="transform">The floor's transform. If set, this will override the floor's elevation.</param>
         /// <param name="openings">An array of openings in the floor.</param>
-        public Floor(Polygon profile, FloorType elementType, double elevation = 0.0, Transform transform = null, List<Opening> openings = null)
+        public Floor(Polygon perimeter, FloorType elementType, double elevation = 0.0, Transform transform = null, List<Opening> openings = null)
         {
-            this.Profile = new Profile(profile);
             if(openings != null)
             {
                 this._openings = openings;
             }
             this.Elevation = elevation;
             this.ElementType = elementType;
             var thickness = elementType.Thickness();
-            this.Transform = transform != null ? transform : new Transform(new Vector3(0, 0, elevation));
-            this.ExtrudeDirection = Vector3.ZAxis;
+            this.Transform = transform ?? new Transform(new Vector3(0, 0, elevation));
+            this.Profile = new Profile(perimeter.Reversed());
```

Comment:
I've thought about this a bit and I'm not sure it works:
- Floors created in this way will now have their perimeters wound in the opposite direction of floors created using the ctor which takes a profile.

Things we can try:
- We could set the profile down and extrude "up", but this would then report the profile below the level of the slab.
- We could set the profile down and extrude "up", but report the perimeter at the correct height.

Reply:
In general, I think we need to revisit the design of these floor constructors. They were a mashup of stuff to support IFC (see the `start` parameter - which doesn't make sense to a user). We'd be better off going straight to floor planes.